core/crypto/sm3.c

```diff
@@ -50,6 +50,16 @@ void sm3_init(struct sm3_context *ctx)
 	ctx->state[7] = 0xB0FB0E4E;
 }
 
+static inline uint32_t rotl(uint32_t val, int shift)
+{
+	if (shift == 0)
```

Comment:
Please add a
``` C
shift &= 0x1F;
```
before this to make sure we're testing the right thing. With this you can also remove the masking of shift further down.